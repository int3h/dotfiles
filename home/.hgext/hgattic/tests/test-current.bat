@echo off

rem setup hgattic in a temp test directory
setlocal
cd ..
cd > tmpfile
set /p atticdir= < tmpfile
erase tmpfile
cd tests
mkdir test
cd test
call hg init
cd .hg
echo [extensions] >tmp
echo hgattic=%atticdir%\attic.py >>tmp

rem now that attic shelves hg format patches all the time a user needs to be
rem configured for tests and datetime information must be provided for the
rem shelve command
echo [ui] >>tmp
echo username=Fred Widget ^<fred@example.com^>>>tmp
echo [defaults] >>tmp
echo attic-shelve = -d "0 0" >>tmp
move tmp hgrc
cd ..

rem replace here with test content
echo ### call hg attic --current
call hg attic --current
echo ### echo a ^> a.txt
echo a > a.txt
echo ### call hg addrem
call hg addrem
echo ### call hg st
call hg st
echo ### call hg shelve --git -q a
call hg shelve --git -q a
echo ### call hg attic -c
call hg attic -c
echo ### call hg unshelve -q
call hg unshelve -q
echo ### call hg attic -c
call hg attic -c
echo ### call hg shelve -r -q -m "commit message"
call hg shelve -r -q -m "commit message"
echo ### call hg attic -c
call hg attic -c
echo ### call hg shelve -r -q -m "another message" -u "test user <asdf@asdf.com>"
call hg shelve -r -q -m "another message" -u "test user <asdf@asdf.com>"
echo ### call hg attic -c
call hg attic -c
echo ### call hg shelve -r -q -d "1 0"
call hg shelve -r -q -d "1 0"
echo ### call hg attic -c
call hg attic -c
echo ### call hg shelve
call hg shelve
echo ### echo b ^> b.txt
echo b > b.txt
echo ### call hg add
call hg add
echo ### call hg shelve -m "new message" b
call hg shelve -m "new message" b
echo ### call hg attic -i a
call hg attic --header a
echo ### call hg attic -i b
call hg attic -d b
rem end test content

rem cleanup
cd ..
rmdir /S /Q test
endlocal
