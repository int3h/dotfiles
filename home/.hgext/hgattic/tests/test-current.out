 ### call hg attic --current
no patch active; default: default
### echo a > a.txt
### call hg addrem
adding a.txt
### call hg st
A a.txt
### call hg shelve --git -q a
### call hg attic -c
no patch active; default: a
### call hg unshelve -q
### call hg attic -c
active patch: a
user: Fred Widget <fred@example.com>
date: 0 0
parent: None
message: None
### call hg shelve -r -q -m "commit message"
### call hg attic -c
active patch: a
user: Fred Widget <fred@example.com>
date: 0 0
parent: None
message: 
commit message
### call hg shelve -r -q -m "another message" -u "test user <asdf@asdf.com>"
### call hg attic -c
active patch: a
user: test user <asdf@asdf.com>
date: 0 0
parent: None
message: 
another message
### call hg shelve -r -q -d "1 0"
### call hg attic -c
active patch: a
user: test user <asdf@asdf.com>
date: 1 0
parent: None
message: 
another message
### call hg shelve
patch a shelved
### echo b > b.txt
### call hg add
adding b.txt
### call hg shelve -m "new message" b
patch b shelved
### call hg attic -i a
user: test user <asdf@asdf.com>
date: 0 0
parent: None
message: 
another message
### call hg attic -i b
user: Fred Widget <fred@example.com>
date: 0 0
parent: None
message: 
new message
